src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -1024,6 +1024,10 @@ public BlittableJsonReaderObject Clone(JsonOperationContext context)
 
             if (Modifications == null)
             {
+                if (forceCloning == false && 
```

Comment:
I do not like that this is only for one case here. Shouldn't this be just after Assert in this method?

Reply:
It make sense to me that cloning the same object on the same context should return `this`.
But was afraid to make this globally, since someone might rely on the fact the cloning gives a completely new instance.

Reply:
I'm not saying that we should not have it, just worried that we are only using this variable when we do not have modifications.

Reply:
This is likely going to cause issues and should be removed.
Scenario, we have a blittable on raw data storage
We clone it explicitly, because we can't modify it and write at the same time.
You have now broken the write, because we read from where we write.

Other issues are about modifying the instance, disposing the old one and then using the new one, etc.

Let's roll back this change.